src/main/java/org/maxgamer/quickshop/integration/lands/LandsIntegration.java

```diff
@@ -67,14 +77,43 @@ public boolean canTradeShopHere(@NotNull Player player, @NotNull Location locati
         return true;
     }
 
-    @Override
-    public void load() {
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onLandsPermissionChanges(LandUntrustPlayerEvent event) {
+        if (!deleteWhenLosePermission) {
+            return;
+        }
+        plugin.getShopManager().getAllShops().forEach((shop -> {
+            if (event.getLand().hasChunk(shop.getLocation().getWorld(), shop.getLocation().getChunk().getX(), shop.getLocation().getChunk().getZ())) {
+                if (event.getTarget().equals(shop.getOwner())) {
+                    plugin.log("[UNTRUSTED DELETE] Shop " + shop + " has been deleted due the owner no-longer have permission in land " + event.getLand().getName());
+                    shop.delete();
+                }
+            }
+        }));
+    }
 
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onLandsMember(PlayerLeaveLandEvent event) {
+        if (!deleteWhenLosePermission) {
+            return;
+        }
+        plugin.getShopManager().getAllShops().forEach((shop -> {
+            if (event.getLand().hasChunk(shop.getLocation().getWorld(), shop.getLocation().getChunk().getX(), shop.getLocation().getChunk().getZ())) {
+                if (event.getLandPlayer().getUID().equals(shop.getOwner())) {
+                    plugin.log("[UNTRUSTED DELETE] Shop " + shop + " has been deleted due the owner no-longer have permission in land " + event.getLand().getName());
+                    shop.delete();
+                }
+            }
+        }));
+    }
 
+    @Override
+    public void load() {
+        Bukkit.getPluginManager().registerEvents(this, plugin);
```

Comment:
It will register event **again** when reload, we should check this.

Reply:
fixed